src/generators.rs

```diff
@@ -983,6 +983,139 @@ pub fn directed_grid_graph(
     })
 }
 
+/// Generate an undirected heavy square graph.
+///
```

Comment:
There is a variant of the heavy square lattice (see Fig 10.b, right-hand side) whose edge case handling (lines 1054-1075 in the present PR) is a bit different. No comment on whether it's worth including as an option, but it is definitely worth mentioning in the docstring.

Reply:
Do you have any suggestion how this can be mentioned in the docstring?

Reply:
The only important thing is to bring to the reader's attention that there is another variant and to disambiguate which you mean. So, something like:

```
NOTE: This function generates the four-frequency variant of the heavy square code. See Fig 10.b left of /1907.09528 , and compare Fig 10.b right.
```

Reply:
The note was added on [`60a150b`](https://github.com/Qiskit/retworkx/pull/313/commits/60a150b0292713220e21a5d00d00935b736fdab9).